src/CMakeLists.txt

```diff
@@ -687,6 +657,32 @@ if(HAVE_ARMV8_CRC)
   list(APPEND ceph_common_deps common_crc_aarch64)
 endif(HAVE_ARMV8_CRC)
 
+if(WITH_DPDK)
+  list(APPEND common_async_dpdk_files
+    msg/async/dpdk/ARP.cc
+    msg/async/dpdk/DPDK.cc
+    msg/async/dpdk/dpdk_rte.cc
+    msg/async/dpdk/DPDKStack.cc
+    msg/async/dpdk/EventDPDK.cc
+    msg/async/dpdk/IP.cc
+    msg/async/dpdk/net.cc
+    msg/async/dpdk/IPChecksum.cc
+    msg/async/dpdk/Packet.cc
+    msg/async/dpdk/TCP.cc
+    msg/async/dpdk/UserspaceEvent.cc
+    msg/async/dpdk/ethernet.cc)
+  add_library(common_async_dpdk STATIC
+    ${common_async_dpdk_files})
+  # to ready the dpdk header files
+  add_dependencies(common_async_dpdk
+    ${DPDK_LIBRARIES})
+  list(APPEND ceph_common_deps common_async_dpdk)
+  target_link_libraries(common_async_dpdk
+    ${DPDK_LIBRARIES})
+  set_target_properties(common_async_dpdk PROPERTIES
+    COMPILE_FLAGS "-march=native -I${DPDK_INCLUDE_DIR}")
```

Comment:
so we can avoid adding "-march=native" globally. this will be a show stopper, if we want to enable DPDK by default.